src/messaging/CHIPReliableMessageContext.cpp

```diff
@@ -0,0 +1,454 @@
+ *  acknowledgement.
+ *
+ *  For internal, debug use only.
+ *
+ */
+bool CHIPReliableMessageContext::ShouldDropAck() const
+{
+    return GetFlag(mFlags, static_cast<uint16_t>(kFlagDropAck));
+}
+
+// Flush the pending Ack
+CHIP_ERROR CHIPReliableMessageContext::RMPFlushAcks()
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    if (IsAckPending())
+    {
+        // Send the acknowledgment as a Common::Null message
+        err = SendCommonNullMessage();
+
+        if (err == CHIP_NO_ERROR)
+        {
+#if defined(DEBUG)
+            ChipLogProgress(ExchangeManager, "Flushed pending ack for MsgId:%08" PRIX32 " to Peer %016" PRIX64, mPendingPeerAckId,
+                            PeerNodeId);
+#endif
+        }
+    }
+
+    return err;
+}
+
+/**
+ *  Get the current retransmit timeout. It would be either the initial or
+ *  the active retransmit timeout based on whether the ExchangeContext has
+ *  an active message exchange going with its peer.
+ *
+ *  @return the current retransmit time.
+ */
+uint32_t CHIPReliableMessageContext::GetCurrentRetransmitTimeout()
+{
+    return (HasRcvdMsgFromPeer() ? mRMPConfig.mActiveRetransTimeout : mRMPConfig.mInitialRetransTimeout);
+}
+
+/**
+ *  Send a Throttle Flow message to the peer node requesting it to throttle its sending of messages.
+ *
+ *  @note
+ *    This message is part of the CHIP Reliable Messaging protocol.
+ *
+ *  @param[in]    pauseTimeMillis    The time (in milliseconds) that the recipient is expected
+ *                                   to throttle its sending.
+ *  @retval  #CHIP_ERROR_INVALID_ARGUMENT               If an invalid argument was passed to this SendMessage API.
+ *  @retval  #CHIP_ERROR_SEND_THROTTLED                 If this exchange context has been throttled when using the
+ *                                                       CHIP reliable messaging protocol.
+ *  @retval  #CHIP_ERROR_WRONG_MSG_VERSION_FOR_EXCHANGE If there is a mismatch in the specific send operation and the
+ *                                                       CHIP message protocol version that is supported. For example,
+ *                                                       this error would be generated if CHIP Reliable Messaging
+ *                                                       semantics are being attempted when the CHIP message protocol
+ *                                                       version is V1.
+ *  @retval  #CHIP_ERROR_NOT_CONNECTED                  If the context was associated with a connection that is now
+ *                                                       closed.
+ *  @retval  #CHIP_ERROR_INCORRECT_STATE                If the state of the exchange context is incorrect.
+ *  @retval  #CHIP_NO_ERROR                             If the CHIP layer successfully sent the message down to the
+ *                                                       network layer.
+ *
+ */
+CHIP_ERROR CHIPReliableMessageContext::RMPSendThrottleFlow(uint32_t pauseTimeMillis)
```

Comment:
can we leave off the "RMP" on methods that the ReliableMessageContext implements to aid readability?

Reply:
done